spec/pi_piper/sysfs/driver_spec.rb

```diff
@@ -0,0 +1,228 @@
+        raise_error(ArgumentError))
+    end
+
+    it 'should raise an error if pin is not exported' do
+      expect { subject.pin_set_trigger(5, :rising) }.to(
+        raise_error(ArgumentError, 'Pin 5 not exported'))
+    end
+  end
+
+  describe '#pin_wait_for' do
+    before(:each) do
+      subject.pin_direction(4, :in)
+      allow(filewatcher).to receive(:watch).and_yield('', :new)
+      allow(FileWatcher).to receive(:new).and_return(filewatcher)
+    end
+
+    it 'should wait for pin to trigger :rising' do
+      expect(subject).to receive(:pin_set_trigger).with(4, :rising)
+      allow(subject).to receive(:pin_read).with(4).and_return(0, 1)
+      expect(subject.pin_wait_for(4, :rising)).to be(true)
+    end
+
+    it 'should wait for pin to trigger :falling' do
+      expect(subject).to receive(:pin_set_trigger).with(4, :falling)
+      allow(subject).to receive(:pin_read).with(4).and_return(1, 0)
+      expect(subject.pin_wait_for(4, :falling)).to be(true)
+    end
+  end
+
+  describe '#close' do
+    it 'should unexport all exported pins' do
+      subject.pin_direction(4, :in)
+      allow(File).to receive(:write).with('/sys/class/gpio/unexport', 4)
+
+      subject.close
+      expect(subject.instance_variable_get(pins)).to be_empty
+    end
+  end
+
+  describe '#unexport' do
+    before(:each) do
+      subject.pin_direction(4, :in)
+    end
+
+    it 'should unexport the pin' do
+      allow(File).to receive(:write).with('/sys/class/gpio/unexport', 4)
+      subject.unexport(4)
+      expect(subject.instance_variable_get(pins)).to be_empty
+    end
+
+    it 'should not export other pins' do
+      allow(File).to receive(:write).with('/sys/class/gpio/unexport', 4)
+      allow(File).to receive(:write).with('/sys/class/gpio/export', 5)
+      allow(File).to(
+        receive(:write).with('/sys/class/gpio/gpio5/direction', :in))
+      subject.pin_direction(5, :in)
+
+      subject.unexport(4)
+      
+      exported_pins = subject.instance_variable_get(pins)
+      expect(exported_pins).to include(5)
+      expect(exported_pins).not_to include(4)
+    end
+  end
+
+  describe '#unexport_all' do
+    it 'should unexport all pins' do
+      allow(File).to receive(:write).with('/sys/class/gpio/export', 5)
+      allow(File).to(
+        receive(:write).with('/sys/class/gpio/gpio5/direction', :in))
+      allow(File).to receive(:write).with('/sys/class/gpio/unexport', 4)
+      allow(File).to receive(:write).with('/sys/class/gpio/unexport', 5)
+
+      subject.pin_direction(4, :in)
+      subject.pin_direction(5, :in)
+
+      subject.unexport_all
+      expect(subject.instance_variable_get(pins)).to be_empty
+    end
+  end
+
+  describe '#exported?' do
+    it 'should return true if pin is exported' do
+      subject.pin_direction(4, :in)
+      expect(subject.exported?(4)).to be(true)
+    end
+
+    it 'should return false if pin is not exported' do
+      expect(subject.exported?(4)).to be(false)
+    end
+  end
+end
```

Comment:
You did a great work by expanding the specs !! you'll have to explain to me why you absolutely want a 100% coverage (as count he gem) ? I'm pretty sure the previous specs did detect the same edge cases.